CHANGELOG.md

```diff
@@ -15,6 +15,9 @@ Increment the:
 
 ## [Unreleased]
 
+* [TEST] Fix failing W3C Trace Context compliance tests [#3115](https://github.com/open-telemetry/opentelemetry-cpp/pull/3115)
```

Comment:
[TEST] may mean it only affects test code, but this fixes W3C trace context API?

Reply:
Yes, I wasn't quite sure about this. This PR touches a lot of aspects, minor API changes, TEST changes and CI updates.
I put [TEST] since the PR was geared toward fixing the failing W3C Trace Context tests.

I'm open to updating it, but not sure what should go here instead.

Reply:
Please use `[API] Comply with W3C Trace Context` then: there are API fixes, which means instrumented code using the api should be rebuilt.

Reply:
Makes sense, I will update the changelog.